### ISSUE TITLE
Modding: Add times limit for a spell

### ISSUE THREAD

**Issue by by003** [vcmi/vcmi#4733]
**Describe your proposal**
In HotA, Eovacius's specialty is creating two clones in the first time, not always.
And in ToW mod, there are some magic staffs which can allow hero to cast free spells.

My suggestion is making a limitation for some special spells.

For example, set a parameter:
`"spellLimitForOneBattle": 1`

**Comment by Alucard648** [comment]
Imagine 7-day cooldown for Town Portal spell.

**Comment by by003** [comment]
Actually Dimension Door is still not configurable for limit times.
Limit times should be used in some modding adventure spells as well, but for one day instead of one battle.
For example, if someone create a spell to gain extra move points, it needs to be used only for one or two times per day.

**Comment by val-gaav** [comment]
DD number of spellcasts  is configurable with this parameter 			"power" : 1  

It is just not documented, but you can get it with /get config  from vcmi in game chat. This will generate all the jsons with current game configs.

**Comment by by003** [comment]
> `number of spell casts per day is configurable with this parameter in json : "power" : 1`

Is this line configured here? However, I'd like to have the function to cast spell per battle. 
Like Eovacius's specialty.
```
// Mandatory. Spell level, value in range 1-5, or 0 for abilities
		"level": 1,

		// Mandatory. Base power of the spell
		"power": 10,
	
		// Mandatory. Default chance for this spell to appear in Mage Guilds
		// Used only if chance for a faction is not set in gainChance field
		"defaultGainChance": 0, 
```

**Comment by GeorgeK1ng** [comment]
Yes, this feature is really needed for [Adventure Spells Pack](https://github.com/vcmi-mods/adventure-spells-pack), because you can cast them multiple times for now, but if you got bonus for week, spell should have defined it's cooldown to prevent multicasting. For example Empathy - Basic needs 1 day colldown, Advanced and Expert 7 days cooldown. According to documentation, this kind of configuration is not possible at the moment. But there is already defined duration ONE_DAY, ONE_WEEK. Now just needs to limit them to cast it only once in casted period.

![image](https://github.com/user-attachments/assets/5559b770-331c-45aa-900b-9babb326eebb)
